src/components/DiagramFrame/utils.ts

```diff
@@ -40,3 +56,37 @@ export function buttonShade(exploreNameSel: string, currentExplore: ILookmlModel
   }
   return undefined
 }
+
+/**
+ * Prepares a list of diagrammable models for the Choose a Model dropdown
+ * @param unfilteredModels - the unprepared list of models
+ */
+export function prepareModelDropdown(unfilteredModels: ILookmlModel[]) {
+  if (unfilteredModels) {
+    return unfilteredModels.filter((d: ILookmlModel)=>{ 
+      return d.explores.length >= 1 && !DIAGRAM_IGNORED_MODELS.includes(d.name)
+    }).map((d: ILookmlModel)=>{
+      return {
+        value: d.name,
+        label: d.label
+      }
+    }).sort((a: SelectOptionProps, b: SelectOptionProps) => a.label < b.label ? -1 : 1)
+  }
+  return []
+}
+
+/**
+ * Prepares a list of diagrammable explores for the Select an Explore list
+ * @param unfilteredModels - the unprepared list of models
+ */
+export function prepareExploreList(currentModel: ILookmlModel) {
+  if (currentModel) {
```

Comment:
Same thing as above

Reply:
Hm this one is a little different bc input is an object and output is an array whereas the above is array to array.